actions/release-publish/action.yml

```diff
@@ -1,4 +1,4 @@
-name: Build and publish release
+name: Tag and publish a release
```

Comment:
I could see the argument that we should in-line this one into the main action as well if it's only called from one place. This is a pretty substantial number of steps, so I could buy an argument that it's better to encapsulate them, but I'll also say that because the GitHub workflows UI shows composite steps as a single entity, it can be difficult to determine what's going on for long workflows like this. That probably becomes less of a problem if this doesn't include the tests (as the output from that currently takes up so much space that it's hard to see what's going on after about the Lucene tests), so it may be fine to leave this as its own thing

Reply:
Yeah, I think your right, and it is cleaner to just pull the actions into the workflow.

I was hoping that it would be much cleaner, like if each job was just a reference to another action, that would be nice, but it didn't end up being that way. `workflows/release.yml` is a bit big, but not crazy.